src/components/Search/SearchRouter/SearchRouterList.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {useCallback} from 'react';
+import * as Expensicons from '@components/Icon/Expensicons';
+import type {SearchQueryJSON} from '@components/Search/types';
+import SelectionList from '@components/SelectionList';
+import SingleIconListItem from '@components/SelectionList/Search/SingleIconListItem';
+import type {ListItemWithSingleIcon, SingleIconListItemProps} from '@components/SelectionList/Search/SingleIconListItem';
+import type {SectionListDataType, UserListItemProps} from '@components/SelectionList/types';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import type {OptionData} from '@libs/ReportUtils';
+import * as SearchUtils from '@libs/SearchUtils';
+import * as Report from '@userActions/Report';
+import ROUTES from '@src/ROUTES';
+
+type ItemWithQuery = {
+    query: string;
+};
+
+type SearchRouterListProps = {
+    currentSearch: SearchQueryJSON | undefined;
+    reportForContextualSearch?: OptionData;
+    recentSearches: ItemWithQuery[] | undefined;
+    recentReports: OptionData[];
+    onRecentSearchSelect: (query: SearchQueryJSON | undefined, shouldAddToRecentSearch?: boolean) => void;
+    closeAndClearRouter: () => void;
+};
+
+function SearchRouterItem(props: UserListItemProps<OptionData> | SingleIconListItemProps<ListItemWithSingleIcon & ItemWithQuery>) {
+    const styles = useThemeStyles();
+
+    if ('item' in props && props.item.reportID) {
+        return (
+            <UserListItem
+                pressableStyle={styles.br2}
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...(props as UserListItemProps<OptionData>)}
+            />
+        );
+    }
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    return <SingleIconListItem {...(props as SingleIconListItemProps<ListItemWithSingleIcon & ItemWithQuery>)} />;
```

Comment:
I'm quite unhappy that we need to do so many `as` casting here. Can we make this better with some smarter type guards?

Reply:
I've tried making it simpler but in the end, that's as pretty as it gets, because you have to cast somewhere.
SelectionList is never used in such case, that there are two different items and OptionData is pretty different from ListType with small inconsistencies(like in one of them alternativeText is `string | undefined | null` and the second is `string | undefined`, and the list of inconsistencies goes on and on).
I'd like to avoid refactoring OptionData and SelectionList types in this PR as it's pretty complicated and I don't want to block our work with this PR, I could then do it in a follow-up PR. WDYT? Also if you have a neat idea how to avoid type casting then let me know because I've run out of ideas.

Reply:
I have a proposal for a slightly alternate types hierarchy for this